pkg/cmd/run/watch/watch.go

```diff
@@ -0,0 +1,228 @@
+	}
+
+	runID := opts.RunID
+	var run *shared.Run
+
+	if opts.Prompt {
+		cs := opts.IO.ColorScheme()
+		runs, err := shared.GetRunsWithFilter(client, repo, 10, func(run shared.Run) bool {
+			return run.Status != shared.Completed
+		})
+		if err != nil {
+			return fmt.Errorf("failed to get runs: %w", err)
+		}
+		if len(runs) == 0 {
+			return fmt.Errorf("found no in progress runs to watch")
+		}
+		runID, err = shared.PromptForRun(cs, runs)
+		if err != nil {
+			return err
+		}
+		// TODO silly stopgap until dust settles and PromptForRun can just return a run
+		for _, r := range runs {
+			if fmt.Sprintf("%d", r.ID) == runID {
+				run = &r
+				break
+			}
+		}
+	} else {
+		run, err = shared.GetRun(client, repo, runID)
+		if err != nil {
+			return fmt.Errorf("failed to get run: %w", err)
+		}
+	}
+
+	if run.Status == shared.Completed {
+		return nil
+	}
+
+	prNumber := ""
+	number, err := shared.PullRequestForRun(client, repo, *run)
+	if err == nil {
+		prNumber = fmt.Sprintf(" #%d", number)
+	}
+
+	if runtime.GOOS == "windows" {
+		opts.IO.EnableVirtualTerminalProcessing()
+	}
+	// clear entire screen
+	fmt.Fprintf(opts.IO.Out, "\x1b[2J")
+
+	annotationCache := map[int][]shared.Annotation{}
+
+	duration, err := time.ParseDuration(fmt.Sprintf("%ds", opts.Interval))
+	if err != nil {
+		return fmt.Errorf("could not parse interval: %w", err)
+	}
+
+	for run.Status != shared.Completed {
+		run, err = renderRun(*opts, client, repo, run, prNumber, annotationCache)
+		if err != nil {
+			return err
+		}
+		time.Sleep(duration)
+	}
+
+	if opts.ExitStatus && run.Conclusion != shared.Success {
+		return cmdutil.SilentError
+	}
+
+	return nil
+}
+
+func renderRun(opts WatchOptions, client *api.Client, repo ghrepo.Interface, run *shared.Run, prNumber string, annotationCache map[int][]shared.Annotation) (*shared.Run, error) {
+	out := opts.IO.Out
+	cs := opts.IO.ColorScheme()
+
+	var err error
+
+	run, err = shared.GetRun(client, repo, fmt.Sprintf("%d", run.ID))
+	if err != nil {
+		return run, fmt.Errorf("failed to get run: %w", err)
+	}
+
+	ago := opts.Now().Sub(run.CreatedAt)
+
+	jobs, err := shared.GetJobs(client, repo, *run)
+	if err != nil {
+		return run, fmt.Errorf("failed to get jobs: %w", err)
+	}
+
+	var annotations []shared.Annotation
+
+	var annotationErr error
+	var as []shared.Annotation
+	for _, job := range jobs {
+		if as, ok := annotationCache[job.ID]; ok {
+			annotations = as
+			continue
+		}
+
+		as, annotationErr = shared.GetAnnotations(client, repo, job)
+		if annotationErr != nil {
+			break
+		}
+		annotations = append(annotations, as...)
+
+		if job.Status != shared.InProgress {
+			annotationCache[job.ID] = annotations
+		}
+	}
+
+	if annotationErr != nil {
+		return run, fmt.Errorf("failed to get annotations: %w", annotationErr)
+	}
+
+	if runtime.GOOS == "windows" {
+		// Just clear whole screen; I wasn't able to get the nicer cursor movement thing working
+		fmt.Fprintf(opts.IO.Out, "\x1b[2J")
```

Comment:
You probably already saw this and tried it out, but incase you didn't https://docs.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences has "Cursor Horizontal Absolute" and "Vertical Line Position Absolute"

Reply:
I tried it and it didn't work :|